src/qibolab/compilers/compiler.py

```diff
@@ -110,20 +107,33 @@ def get_sequence(self, gate, platform):
         """
         # get local sequence for the current gate
         rule = self[type(gate)]
-        if isinstance(gate, (gates.M, gates.Align)):
+
+        natives = platform.parameters.native_gates
+
+        if isinstance(gate, (gates.M)):
+            qubits = [
+                natives.single_qubit[platform.get_qubit(q).name] for q in gate.qubits
+            ]
+            return rule(gate, qubits)
```

Comment:
I don't fully like all these `if`s as it's a bit against of the idea of having the dict of rules (which can also be modified by user) and delegate as much possible to that. However, I have not spend time to thought in detail for a better mechanism so I don't want to just complain without providing any alternatives. Maybe a simple one, without thinking much, would be to pass the whole `platform` to each rule, as it was before.

If you also don't fully like it, maybe we could open an issue and try to improve in the future. I am not sure if it's for 0.2 as the compiler is not the usual user interfacing kind, but could also be thought as interface for another kind of users.

Reply:
> I don't fully like all these `if`s as it's a bit against of the idea of having the dict of rules (which can also be modified by user) and delegate as much possible to that. However, I have not spend time to thought in detail for a better mechanism so I don't want to just complain without providing any alternatives. Maybe a simple one, without thinking much, would be to pass the whole `platform` to each rule, as it was before.

I'm not fully convinced, but I also don't fully like the idea of passing everything down below.
This is truly the trade-off I arrived, in which I'm essentially dividing in 2+ categories: single-qubit, two-qubit, special handling.

Since the special is a catch-all category, maybe it's worth to just pass everything down just in this case.
For the other two categories, it should be pretty clear what to do. But unfortunately you can isolate them only if you remove the special ones before, since `M` and `Align` may also apply to one and two qubits. Maybe we could have a list of those that should be treated as special, and expose it to the user for extensions.

> If you also don't fully like it, maybe we could open an issue and try to improve in the future. I am not sure if it's for 0.2 as the compiler is not the usual user interfacing kind, but could also be thought as interface for another kind of users.

In principle, the compiler is only used by the user for extensions. Since we do not believe that extensions are immediately needed, we could keep it private, and publish only at a later stage.